<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>My Portfolio</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Kevin Boyd | Data Science & Bioinformatics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="About_Me.html">About Me</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Applied Bioinformatics Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="DeepLearning_CellStageClassification.html">Deep Learning Classification</a>
    </li>
    <li>
      <a href="Nuclear_Colocalization.html">Nuclear Colocalization</a>
    </li>
    <li>
      <a href="CutandTagAnalysis.html">Cut and Tag Analysis</a>
    </li>
    <li>
      <a href="ReplicatePeakAnalysis.html">Cut and Tag Replicate Peaks</a>
    </li>
    <li>
      <a href="ChEC-seq_Analysis.html">ChEC-Seq Analysis</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Training Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="DDS_Project1.html">Budweiser Case Study</a>
    </li>
    <li>
      <a href="CLT2.html">Central Limit Theorem</a>
    </li>
    <li>
      <a href="DDS_Project2_Final.html">Attrition Case Study</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">My Portfolio</h1>
<h4 class="author"><p>Author: Kevin
Boyd<br><a href='https://github.com/kevinboyd76'><img src='https://img.shields.io/badge/GitHub-kevinboyd76-181717?logo=github&style=flat&logoColor=white' alt='GitHub Badge'></a></p></h4>
<h4 class="date">Last Updated: 7/1/2025</h4>

</div>


<p><img src="images/hello_world.jpg" /></p>
<p>My name is <strong>Kevin Boyd</strong>! I’m a data scientist and
bioinformatician with a passion for utilizing complex data into
actionable insights. I currently live in Oklahoma and work as a Senior
Bioinformatic Analyst. I specialize in statistical modeling, data
visualization, image analysis, and building end-to-end data pipelines
with R and Python.</p>
<hr />
<div id="overview" class="section level1">
<h1>Overview</h1>
<p>This portfolio highlights projects that demonstrate my past work in
data science and bioinformatics. It includes real world analysis and NGS
pipelines from my professional bioinformatic work as well as training
projects completed during my graduate studies at SMU.</p>
<p>In each section I have provided an overview of the project and a link
to the GitHub repository. I have also included links to Shiny
applications, case study videos, and publications when applicable. These
examples show end to end workflows and are focused on translating data
into actionable insights.</p>
<hr />
</div>
<div id="applied-bioinformatics-projects" class="section level1">
<h1>Applied Bioinformatics Projects:</h1>
<p>This section showcases bioinformatics projects I’ve contributed to
professionally, focused on genomic data analysis, image analysis, and
end to end workflow development. These projects support grant proposals
and publications, leveraging input from domain experts to design
reproducible and scalable workflows tailored to biological research.</p>
<hr />
<div id="deep-learning-image-classification" class="section level3">
<h3>Deep Learning Image Classification</h3>
<p><strong>Goal</strong>: Classify human cells into five distinct
patterns representing S-phase progression using a convolutional neural
network (CNN).</p>
<ul>
<li>Developed deep learning pipeline for image segmentation and
classification<br />
</li>
<li>Preprocessed raw microscopy data and generated labeled training
sets<br />
</li>
<li>Built CNN-based models to predict nuclear patterns across
timepoints<br />
</li>
<li>Integrated model into publication and GitHub-hosted analysis
toolkit</li>
</ul>
<p><a href="DeepLearning_CellStageClassification.html">Deep Learning
Classification Tab</a><br />
<a
href="https://github.com/kevinboyd76/Automated_S-phase_Image_Classification">GitHub
Repo</a></p>
<hr />
</div>
<div id="nuclear-colocalization-analysis" class="section level3">
<h3>Nuclear Colocalization Analysis</h3>
<p><strong>Goal</strong>: Quantify colocalization between nuclear
proteins using multi-channel fluorescence microscopy.</p>
<ul>
<li>Designed reproducible image analysis workflow using Python and
<code>skimage</code><br />
</li>
<li>Calculated Pearson’s and Manders’ correlation coefficients for
nuclear ROIs<br />
</li>
<li>Enabled batch processing across experimental conditions<br />
</li>
<li>Supported statistical reporting and figure generation for
publication</li>
</ul>
<p><a href="Nuclear_Colocalization.html">Nuclear Colocalization
Tab</a><br />
<a href="https://github.com/CS-Cobre-Help/Nuclear_Colocalization">GitHub
Repo</a></p>
<hr />
</div>
<div id="cuttag-analysis-pipeline" class="section level3">
<h3>CUT&amp;Tag Analysis Pipeline</h3>
<p><strong>Goal</strong>: Develop a modular CUT&amp;Tag data analysis
pipeline for chromatin profiling in human cells using snakemake. This
pipeline uses fastqs (raw sequencing files) as input.</p>
<ul>
<li>Designed reproducible pipeline for read alignment, peak calling, and
signal normalization<br />
</li>
<li>Generated QC plots and coverage tracks for visualization in
IGV<br />
</li>
<li>Created summary statistics for sample performance and antibody
enrichment<br />
</li>
<li>Documented pipeline for lab use and shared as part of reproducible
research initiatives</li>
</ul>
<p><a href="CutandTagAnalysis.html">Cut&amp;Tag Analysis Tab</a><br />
<a
href="https://github.com/JK-Cobre-Help/CutandTag_Analysis_Snakemake">GitHub
Repo</a></p>
<hr />
</div>
<div id="cuttag-replicate-peak-integration" class="section level3">
<h3>CUT&amp;Tag Replicate Peak Integration</h3>
<p><strong>Goal</strong>: Integrate biological replicates in CUT&amp;Tag
experiments to define consensus peaks. This pipeline performs downstream
analysis using genome aligned files as input.</p>
<ul>
<li>Merged aligned BAM files across replicates to increase
reproducibility<br />
</li>
<li>Identified reproducible peaks using a peak overlap threshold<br />
</li>
<li>Generated Euler plots and summary metrics for peak consistency<br />
</li>
<li>Generate heatmaps to show signal over specific genomic
intervals<br />
</li>
<li>Provided reproducible scripts for future experiments</li>
</ul>
<p><a href="ReplicatePeakAnalysis.html">Replicate Peak Analysis
Tab</a><br />
<a
href="https://github.com/JK-Cobre-Help/CutandTag_ReplicatePeak_Analysis">GitHub
Repo</a></p>
<hr />
</div>
<div id="chec-seq-analysis" class="section level3">
<h3>ChEC-seq Analysis</h3>
<p><strong>Goal</strong>: Build a reproducible pipeline for analyzing
ChEC-seq data from S. cerevisiae and S. pombe.</p>
<ul>
<li>Aligned primary and spike-in reads to dual reference genomes<br />
</li>
<li>Performed spike-in and CPM normalization for cross-sample
comparison<br />
</li>
<li>Generated signal tracks and enrichment profiles across genomic
features<br />
</li>
<li>Developed modular, Snakemake-based workflow for reproducibility</li>
</ul>
<p><a href="ChEC-seq_Analysis.html">ChEC-seq Analysis Tab</a><br />
<a href="https://github.com/RD-Cobre-Help/ChEC-Seq_Analysis">GitHub
Repo</a></p>
<hr />
</div>
</div>
<div id="training-projects" class="section level1">
<h1>Training Projects:</h1>
<p>This section highlights specific projects that I completed during my
graduate training in data science at SMU. These projects cover
statistical modeling, classification, clustering, and visualization
using R, Python, and Shiny.</p>
<div id="shiny-apps" class="section level3">
<h3>Shiny Apps</h3>
<ul>
<li><a
href="https://kevinboyd76.shinyapps.io/k-NN_Classifier/">k-NN_Classifier</a></li>
</ul>
<hr />
</div>
<div id="budweiser-marketing-case-study" class="section level3">
<h3>Budweiser Marketing Case Study</h3>
<p><strong>Goal</strong>: Identify errors and recognize patterns that
are present in the beers and breweries data sets given to us by
Budweiser. We also want to determine if there is a relationship between
ABV (alcohol by volume) and IBU (international bitterness units).</p>
<ul>
<li>Explored k-means clustering and PCA</li>
<li>Visualized brand clusters in 2D PCA space</li>
<li>Presented results in an academic setting</li>
</ul>
<p><a href="DDS_Project1.html">View Case Study Tab</a><br />
<a href="https://youtu.be/fjBUYcG8ZlI">Budweiser Case Study
Video</a></p>
<hr />
</div>
<div id="employee-attrition-prediction" class="section level3">
<h3>Employee Attrition Prediction</h3>
<p><strong>Goal</strong>: Identify factors that predict employee
attrition based on statistical analysis. Finally build a classification
model that can predict attrition and salary.</p>
<ul>
<li>Trained Naive Bayes, k-NN, and linear regression models</li>
<li>Evaluated classification models on accuracy, precision, recall</li>
<li>Evaluated linear regression model on RMSE and p-value.</li>
</ul>
<p><a href="DDS_Project2_Final.html">View Attrition Tab</a><br />
<a href="https://youtu.be/MuLEzt19qhw">Attrition Case Study
Video</a></p>
<hr />
</div>
<div id="central-limit-theorem-simulation" class="section level3">
<h3>Central Limit Theorem Simulation</h3>
<p><strong>Goal</strong>: Visualize how the CLT emerges with increasing
sample sizes.</p>
<ul>
<li>Simulated 10,000 draws from various distributions</li>
<li>Showed how sampling distributions approach normality</li>
<li>Used animated ggplot2 to demonstrate convergence</li>
</ul>
<p><a href="CLT2.html">View CLT Tab</a></p>
<hr />
<div id="notes" class="section level4">
<h4>Notes</h4>
<p>These projects were developed as part of SMU’s Data Science graduate
program (2021–2022) and reflect foundational skill development across
applied machine learning, statistics, and data storytelling.</p>
</div>
</div>
<div id="contact" class="section level2">
<h2>Contact</h2>
<p>Feel free to look around and reach out if you would like to connect!
You can contact me by email at <a href="mailto:kevinboyd76@gmail.com"
class="email">kevinboyd76@gmail.com</a></p>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
