---
title: "My Portfolio"
author: |
  Author: Kevin Boyd<br><a href='https://github.com/kevinboyd76'><img src='https://img.shields.io/badge/GitHub-kevinboyd76-181717?logo=github&style=flat&logoColor=white' alt='GitHub Badge'></a>
date: "Last Updated: 7/1/2025"
output: html_document
---

![](images/hello_world.jpg)  

My name is **Kevin Boyd**! I'm a data scientist and bioinformatician with a passion for utilizing complex data into actionable insights. I currently live in Oklahoma and work as a Senior Bioinformatic Analyst. I specialize in statistical modeling, data visualization, image analysis, and building end-to-end data pipelines with R and Python.

 ---

# Overview  

This portfolio highlights projects that demonstrate my past work in data science and bioinformatics. It includes real world analysis and NGS pipelines from my professional bioinformatic work as well as training projects completed during my graduate studies at SMU.

In each section I have provided an overview of the project and a link to the GitHub repository. I have also included links to Shiny applications, case study videos, and publications when applicable. These examples show end to end workflows and are focused on translating data into actionable insights.

 ---

# Applied Bioinformatics Projects:  

This section showcases bioinformatics projects I’ve contributed to professionally, focused on genomic data analysis, image analysis, and end to end workflow development. These projects support grant proposals and publications, leveraging input from domain experts to design reproducible and scalable workflows tailored to biological research.  

 ---

### Deep Learning Image Classification

**Goal**: Classify human cells into five distinct patterns representing S-phase progression using a convolutional neural network (CNN).  

- Developed deep learning pipeline for image segmentation and classification  
- Preprocessed raw microscopy data and generated labeled training sets  
- Built CNN-based models to predict nuclear patterns across timepoints  
- Integrated model into publication and GitHub-hosted analysis toolkit  

[Deep Learning Classification Tab](DeepLearning_CellStageClassification.html)  
[GitHub Repo](https://github.com/kevinboyd76/Automated_S-phase_Image_Classification)

 ---

### Nuclear Colocalization Analysis

**Goal**: Quantify colocalization between nuclear proteins using multi-channel fluorescence microscopy.

- Designed reproducible image analysis workflow using Python and `skimage`  
- Calculated Pearson’s and Manders’ correlation coefficients for nuclear ROIs  
- Enabled batch processing across experimental conditions  
- Supported statistical reporting and figure generation for publication  

[Nuclear Colocalization Tab](Nuclear_Colocalization.html)  
[GitHub Repo](https://github.com/CS-Cobre-Help/Nuclear_Colocalization)

 ---

### CUT&Tag Analysis Pipeline

**Goal**: Develop a modular CUT&Tag data analysis pipeline for chromatin profiling in human cells using snakemake. This pipeline uses FASTQ (raw sequencing files) as input and produces QC plots, BAM files, BigWigs (scaled and unscaled), and calls peaks with MACS2.  

- Pipeline used for read alignment, peak calling, and signal normalization 
- Specific genome used for alignment can easily be changed in config.yml
- Generate alignment statistics to produce QC plots and create coverage tracks for visualization in IGV  
- Documented pipeline for lab use to improve reproducibility and assigned a DOI for publication and broader accessibility

[Cut&Tag Analysis Tab](CutandTagAnalysis.html)  
[GitHub Repo](https://github.com/JK-Cobre-Help/CutandTag_Analysis_Snakemake)

 ---

### CUT&Tag Replicate Peak Analysis

**Goal**: Combine biological replicates in CUT&Tag experiments to create consensus peak files. This pipeline performs downstream analysis on consensus peak sets using genome aligned BAM files as input.

- Merged aligned BAM files across replicates  
- Identified reproducible peaks using a peak overlap threshold set in the config.yml (eg: 2/3 samples overlap with merged peak set)   
- Generate Euler plots and summary metrics for peak consistency  
- Compares overlap of consensus peaks for each condition provided in the samples.csv   
- Generate heatmaps to show signal over specific genomic intervals   
- Documented pipeline for lab use to improve reproducibility and assigned a DOI for publication and broader accessibility  

[Replicate Peak Analysis Tab](ReplicatePeakAnalysis.html)  
[GitHub Repo](https://github.com/JK-Cobre-Help/CutandTag_ReplicatePeak_Analysis)

 ---

### ChEC-seq Analysis

**Goal**: Build a reproducible pipeline for analyzing ChEC-seq data aligned to a primary genome (eg: S. cerevisiae) and a spike-in genome (eg: S. pombe). This pipeline takes FASTQ files as input and produces multiple coverage tracks including raw, CPM, and spike-in normalized.  

- Align primary and spike-in reads to specific reference genomes set in the config.yml   
- Performed spike-in and CPM normalization for cross-sample comparison  
- Generated signal tracks and enrichment profiles across genomic features  
- Produce QC plots that show alignment statistics and fragment lengths  
- Documented pipeline for lab use to improve reproducibility and assigned a DOI for publication and broader accessibility  

[ChEC-seq Analysis Tab](ChEC-seq_Analysis.html)  
[GitHub Repo](https://github.com/RD-Cobre-Help/ChEC-Seq_Analysis)

 ---

# Training Projects:  

This section highlights specific projects that I completed during my graduate training in data science at SMU. These projects cover statistical modeling, classification, clustering, and visualization using R, Python, and Shiny.

### Shiny Apps
- [k-NN_Classifier](https://kevinboyd76.shinyapps.io/k-NN_Classifier/)

 ---

### Budweiser Marketing Case Study

**Goal**:  Identify errors and recognize patterns that are present in the beers and breweries data sets given to us by Budweiser. We also want to determine if there is a relationship between ABV (alcohol by volume) and IBU (international bitterness units).

- Explored k-means clustering and PCA
- Visualized brand clusters in 2D PCA space
- Presented results in an academic setting  

[View Case Study Tab](DDS_Project1.html)  
[Budweiser Case Study Video](https://youtu.be/fjBUYcG8ZlI)  

 ---

### Employee Attrition Prediction

**Goal**: Identify factors that predict employee attrition based on statistical analysis. Finally build a classification model that can predict attrition and salary.

- Trained Naive Bayes, k-NN, and linear regression models
- Evaluated classification models on accuracy, precision, recall
- Evaluated linear regression model on RMSE and p-value.  

[View Attrition Tab](DDS_Project2_Final.html)  
[Attrition Case Study Video](https://youtu.be/MuLEzt19qhw)  

 ---

### Central Limit Theorem Simulation

**Goal**: Visualize how the CLT emerges with increasing sample sizes.

- Simulated 10,000 draws from various distributions
- Showed how sampling distributions approach normality
- Used animated ggplot2 to demonstrate convergence  

[View CLT Tab](CLT2.html)

 ---

#### Notes

These projects were developed as part of SMU's Data Science graduate program (2021–2022) and reflect foundational skill development across applied machine learning, statistics, and data storytelling.

## Contact
Feel free to look around and reach out if you would like to connect! You can contact me by email at kevinboyd76@gmail.com